### PR TITLE
SG-35559 Use Enum for pyside2/6. Standardize osutils.

### PR DESCRIPTION
**Pull request by carlos-villavicencio-adsk**
- Use Enum for pyside2/6
    - QtGui.QTextCursor.End
    - QtGui.QTextCursor.StartOfLine
    - [QtGui.QItemSelectionModel.SelectCurrent](https://doc.qt.io/qtforpython-6.5/PySide6/QtCore/QItemSelectionModel.html)
- Removed `WA_MacNoShadow` [QTBUG-94052](https://bugreports.qt.io/browse/QTBUG-94052?focusedCommentId=568642)
- Refactor `ProjectCommunication`, `SiteCommunication`, and `CommunicationBase` to don't receive `engine` argument in constructor and prevent multiple inheritance issue with PySide6
- `accepted` attribute to prevent `AttributeError: 'SetupProjectWizard' object has no attribute 'Accepted'. Did you mean: 'accepted'?` error
- Standardize `osutils` module to make it platform agnostic.
- Add explicit alignment for footer's push buttons. Otherwise user button looks misaligned.
- Related PR: https://github.com/shotgunsoftware/tk-framework-adminui/pull/48

### CODE REVIEW

**Review by julien-lang — APPROVED** [review]
LGTM!

File: python/tk_desktop/systray.py [hunk 1]
```diff
@@ -81,9 +81,6 @@ def deactivate(self, deactivated):
     def __init__(self, parent=None):
         QtGui.QMainWindow.__init__(self, parent)
 
-        if sys.platform == "darwin":
-            self.setAttribute(QtCore.Qt.WA_MacNoShadow)
```
Comment (julien-lang) [inline python/tk_desktop/systray.py:85]: Why do we remove this?

File: python/tk_desktop/desktop_engine_site_implementation.py [hunk 2]
```diff
@@ -37,7 +37,9 @@
 class DesktopEngineSiteImplementation(object):
     def __init__(self, engine):
 
-        self.site_comm = SiteCommunication(engine)
+        # self.site_comm = SiteCommunication(engine)
```
Comment (julien-lang) [inline python/tk_desktop/desktop_engine_site_implementation.py:40]: comment to remove.

File: python/tk_desktop/systray.py (hunk 1 above)
Comment (carlos-villavicencio-adsk) [reply python/tk_desktop/systray.py:85]: This has been removed from [Qt5](https://doc.qt.io/qt-5/macos-issues.html) to [Qt6](https://doc.qt.io/qt-6/macos-issues.html). Reference https://bugreports.qt.io/browse/QTBUG-94052?focusedCommentId=568642

Comment (carlos-villavicencio-adsk) [reply python/tk_desktop/systray.py:85]: Description updated

**Review by julien-lang — APPROVED** [review]

File: python/tk_desktop/site_communication.py [hunk 3]
```diff
@@ -28,12 +28,12 @@ class SiteCommunication(QtCore.QObject, CommunicationBase):
     proxy_closing = QtCore.Signal()
     proxy_created = QtCore.Signal()
 
-    def __init__(self, engine):
+    def __init__(self):
         """
         :param engine: Toolkit engine.
```
Comment (eduardoChaucaGallegos) [inline python/tk_desktop/site_communication.py:33]: This param docstring is still correct?

Comment (eduardoChaucaGallegos) [inline python/tk_desktop/site_communication.py:33]: This param docstring is still correct?

**Review by julien-lang — APPROVED** [review]

**Review by julien-lang — APPROVED** [review]

**Pull request merged**